src/main/scala/rocket/ICache.scala

```diff
@@ -208,66 +438,134 @@ class ICacheModule(outer: ICache) extends LazyModuleImp(outer)
     v
   }
 
+/**  init tag SRAM, indexed with virtual memory,
+ *   content with `refillError ## tag[19:0]` after ECC
+ * */
   val tag_array  = DescribedSRAM(
     name = "tag_array",
     desc = "ICache Tag Array",
     size = nSets,
     data = Vec(nWays, UInt(width = tECC.width(1 + tagBits)))
   )
-
   val tag_rdata = tag_array.read(s0_vaddr(untagBits-1,blockOffBits), !refill_done && s0_valid)
+  /** register indicates the ongoing GetAckData transaction is corrupted. */
   val accruedRefillError = Reg(Bool())
+  /** wire indicates the ongoing GetAckData transaction is corrupted. */
   val refillError = tl_out.d.bits.corrupt || (refill_cnt > 0 && accruedRefillError)
   when (refill_done) {
     // For AccessAckData, denied => corrupt
+    /** data written to [[tag_array]].
+     *  ECC encoded `refillError ## refill_tag`*/
     val enc_tag = tECC.encode(Cat(refillError, refill_tag))
     tag_array.write(refill_idx, Vec.fill(nWays)(enc_tag), Seq.tabulate(nWays)(repl_way === _))
 
     ccover(refillError, "D_CORRUPT", "I$ D-channel corrupt")
   }
+  // notify CPU, I$ has corrupt.
   io.errors.bus.valid := tl_out.d.fire() && (tl_out.d.bits.denied || tl_out.d.bits.corrupt)
   io.errors.bus.bits  := (refill_paddr >> blockOffBits) << blockOffBits
 
+  /** true indicate this cacheline is valid,
+    * indexed by (wayIndex ## setIndex)
+    * after refill_done and not FENCE.I, (repl_way ## refill_idx) set to true.
+    */
   val vb_array = Reg(init=Bits(0, nSets*nWays))
   when (refill_one_beat) {
     accruedRefillError := refillError
+    // @todo seems comments are not corresponding to codes?
```

Comment:
What do you mean here? The comment seems to be correct

Reply:
It seems correct for me as well. I will fix it.